air/src/air/mod.rs

```diff
@@ -579,25 +582,59 @@ pub trait Air: Send + Sync {
         R: RandomCoin<BaseField = Self::BaseField>,
     {
         let mut t_coefficients = Vec::new();
-        for _ in 0..self.trace_info().width() {
-            t_coefficients.push(public_coin.draw()?);
+        match self.context().options.batching_type_deep() {
+            BatchingType::Linear => {
+                for _ in 0..self.trace_info().width() {
+                    t_coefficients.push(public_coin.draw()?);
+                }
+
+                let mut c_coefficients = Vec::new();
+                for _ in 0..self.context().num_constraint_composition_columns() {
+                    c_coefficients.push(public_coin.draw()?);
+                }
+
+                let lagrange_cc = if self.context().has_lagrange_kernel_aux_column() {
+                    Some(public_coin.draw()?)
+                } else {
+                    None
+                };
+
+                Ok(DeepCompositionCoefficients {
+                    trace: t_coefficients,
+                    constraints: c_coefficients,
+                    lagrange: lagrange_cc,
+                })
+            },
+            BatchingType::Algebraic => {
+                let mut t_coefficients = Vec::new();
+                let alpha: E = public_coin.draw()?;
+                t_coefficients
+                    .extend_from_slice(&get_power_series(alpha, self.trace_info().width()));
```

Comment:
`get_power_series()` would return `1` as the first entry in the series - is that an issue?

Reply:
Yeah, that is the intended behavior. Namely if we have two codewords we want to batch then the batched word will be $\alpha^0 \cdot f_0 + \alpha^1 \cdot f_1$ i.e. $f_0 + \alpha\cdot f_1$